Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -1,52 +1,9 @@
 #ifndef CURRENTDEPOSITION_H_
 #define CURRENTDEPOSITION_H_
 
-using namespace amrex;
-
-// Compute shape factor and return index of leftmost cell where 
-// particle writes.
-// Specialized templates are defined below for orders 1, 2 and 3.
-template <int depos_order>
-AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-int compute_shape_factor(Real* const sx, Real xint) {return 0;};
-
-// Compute shape factor for order 1.
-template <>
-AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-int compute_shape_factor <1> (Real* const sx, Real xmid){
-    int j = (int) xmid;
-    Real xint = xmid-j;
-    sx[0] = 1.0 - xint;
-    sx[1] = xint;
-    return j;
-}
+#include "ShapeFactors.H"
 
-// Compute shape factor for order 2.
-template <>
-AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-int compute_shape_factor <2> (Real* const sx, Real xmid){
-    int j = (int) (xmid+0.5);
-    Real xint = xmid-j;
-    sx[0] = 0.5*(0.5-xint)*(0.5-xint);
-    sx[1] = 0.75-xint*xint;
-    sx[2] = 0.5*(0.5+xint)*(0.5+xint);
-    // index of the leftmost cell where particle deposits
-    return j-1;
-}
-
-// Compute shape factor for order 3.
-template <>
-AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
-int compute_shape_factor <3> (Real* const sx, Real xmid){
-    int j = (int) xmid;
-    Real xint = xmid-j;
-    sx[0] = 1.0/6.0*(1.0-xint)*(1.0-xint)*(1.0-xint);
-    sx[1] = 2.0/3.0-xint*xint*(1-xint/2.0);
-    sx[2] = 2.0/3.0-(1-xint)*(1-xint)*(1.0-0.5*(1-xint));
-    sx[3] = 1.0/6.0*xint*xint*xint;
-    // index of the leftmost cell where particle deposits
-    return j-1;
-}
+using namespace amrex;
```

Comment:
just for the sake of code cleanness, we should avoid using other namespaces in headers. this can pollute global classes and lead to hard-to-grasp clashes. also, it's not really needed since the classes below are mostly pre-fixed if they come from the amrex namespace. but we can take a look at this another time.

Reply:
Mostly pre-fixed?? `Real` is `amrex::Real`, so I change it *everywhere*.
Joking aside, you're right, I did the modifications.

Reply:
Oh, I was only checking function signatures :)

Be aware that writing using namespace; is totally fine within scopes, e.g. at the beginning of a function body. It's just advisable to avoid it in global scope that get carried on via includes.